### DISCUSSION TITLE
[Question] What exactly is the difference between refetch and refresh?

### DISCUSSION THREAD

**Discussion by tafelnl** [Akryum/vue-apollo#939] [Q&A]
Title speaks for itself. When I do a `refetch`  on a query it will not change a possible `$apollo.queries[queryName].loading === false` to `true`. In contrary this **is** the case with `refresh`. So what exactly is the technical difference between the two.

https://github.com/vuejs/vue-apollo/issues/36#issuecomment-321524827 states that refetch is the new refresh. But I do not believe that to be correct since they behave different and are also stated separately in the docs. ([refetch](https://apollo.vuejs.org/api/smart-query.html#refetch), [refresh](https://apollo.vuejs.org/api/smart-query.html#refresh)).

I would love to hear the exact difference between the two :)

**Comment by Dravere** [comment]
So I was interested in this as well and went a bit digging in the code. You probably might have already found the answer to this, but in case someone else might find this issue like me, here we go. Disclaimer: I'm just starting with vue-apollo and apollo and thus I'm no expert.

As far as I can tell `refetch` just calls `refetch` on the apollo client. Thus it just issues the query again via apollo in the way it created it in the beginning. Optionally changing the query variables with the one provided via the argument. As far as I can tell this seems to be a permanent change of the variables. The next call to `refetch` with no argument will use the variables last given to `refetch`.

The refresh function on the other hand stops any on-going querying. And then partially recreates the query and executes it. This means that if you use a reactive query definition it will execute that function again before fetching the data. The same goes for reactive parameters.

If a result can be loaded fast enough via apollo the loading property never will be set to true. Since `refetch` is much more lightweight my assumption is that your server responds fast enough and thus you never see that changing to `true`. And since `refresh` is much more heavy and basically has to recreate the query on apollo it probably is just slow enough that the loading state is set to `true`.
